libs/shared/form-fields/src/lib/CompanySearchController/CompanySearchController.tsx

```diff
@@ -94,6 +111,34 @@ export const CompanySearchController: FC<Props> = ({
     return options || []
   }
 
+  const getIsatNumbers = (nationalId: string) => {
+    if (!shouldIncludeIsatNumber) return ''
+
+    const companies: Array<any> = data?.companyRegistryCompanies?.data ?? []
+    if (companies.length === 0) return ''
+
+    const filteredCompany = companies.filter((company) => {
+      if (company.nationalId === nationalId) return true
+      return false
+    })
+
+    if (filteredCompany.length === 0) return ''
+    const vats: Array<any> = filteredCompany[0].companyInfo.vat
+    let isat = ''
+    vats?.map((v) => {
+      if (!v.dateOfDeregistration) {
+        ;(v.classification as Array<any>)?.map((c) => {
+          if (c.type === 'Aðal') {
+            isat = `${c.number} - ${c.name}`
+          }
+          return c
+        })
+      }
+      return v
+    })
+    return isat
+  }
```

Comment:
The search for the main ISAT can be improved to return when you find it
```suggestion
    for (let v of vats) {
      if (!v.dateOfDeregistration) {
        const classification = v.classification.find(c => c.type === 'Aðal')
        if (classification) {
          return `${c.number} - ${c.name}`
        }
      }
    }
    return ''
  }
```